specs/signing-and-verification-workflow.md

```diff
@@ -70,3 +70,46 @@ The user wants to pull an OCI artifact only if they are signed by a trusted publ
            If all signature artifact descriptors have already been processed, fail the signature verification and exit.
 1. **Get OCI artifact:** Using the verified digest, download the OCI artifact.
    This step is not in the purview of Notary Project.
+
+## Blob signing workflow
+
+The user wants to sign an arbitrary blob with a detached signature.
+
+### Signing Prerequisites
+
+- User has access to the signing certificate and private key or a remote signing service through a notation plug-in.
+
+### Signing Steps
+
+1. **Generate signature:** Using notation CLI or any other compliant signing tool, sign the blob. The signing tool should follow the following guideline.
+    1. Construct the blob payload as defined in [`signature specification`](./signature-specification.md#payload)
+    1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+    1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+    1. Generate signature.
+        1. Generate signature using a signature format specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature payload.
+        1. If the user wants to timestamp the signature, obtain an [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant timestamp for the signature generated in the previous step. Otherwise, continue to the next step.
+            1. Verify that the timestamp signing certificate satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+            1. Verify that the timestamp signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+        1. Embed timestamp to the signature envelope.
+1. **Save the signature envelope:** Save the signature envelope generated in the previous step to a file. File extension should be the original blob file name plus `.sig.jws` for JWS signatures and `.sig.cose` for COSE signatures.
```

Comment:
.sig is actually not a popular extension for digital signatures, unlike json. Different tools use different extensions based on their requirements. I think for our usecase, it is okay to go with either .sig.jws or .jws.sig. Both seem fine to me. Unless we have a string reason to flip the order, lets stick to what we have.

Reply:
I more inclined towards consistency with trust policy file name unless we have a usecase/requirement that doesn't works with `.sig`. Also .sig for signature is a popular extension used by many tools such as cosign, [in email signature](https://removefile.com/sig-file-extension/),

Reply:
Consistency with trust policies seems like a weak argument since json files are meant to be read and understood by various tools while signature files are meant for Notation only (since only Notation can understand the signature format we have). Again, I'm fine either ways. I don't have a strong opinion here. @shizhMSFT do you have a preference here?

Reply:
TBH, I'd prefer `.sig` for all JWS and COSE signatures (and use other mechanisms like PEM or file magic numbers) but I'm fine that we distinguish them using file extension `.sig.jws` and `.sig.cose`.

In the case we are going the way of `sig.jws` and `sig.cose`, I'd prefer as it is. Taking `.tar.gz` as an example, `foo.tar.gz` is a gzip file containing the content of `foo.tar`. Hence, a JWS/COSE file file containing the content of a Notary Project signature is better to have the extension name of `.sig.jws` / `.sig.cose` instead of `.jws.sig` and `.cose.sig`.

In the case we are going the way of using `.sig` for all JWS and COSE signatures eventually, I'd like to say that `.jws.sig` and `.cose.sig` are better since both `.jws.sig` and `.cose.sig` are `.sig` files.

Note: email signatures use `.p7s` (pkcs#7/CMS) for the signature extension.

Reply:
Typically speaking, users think the token after the last dot as the file extension and prefer not to modify it. In that sense, keeping .sig.jws and .sig.cose yields better results than .jws.sig or .cose.sig (here, I'm referring to notation not needing to guess the format by trying all possibilities as a better result)